packages/web/src/ssr/+onRenderClient.tsx

```diff
@@ -34,19 +31,20 @@ export default async function render(
   if (HYDRATE_CLIENT) {
     hydrateRoot(
       document.getElementById('root') as HTMLElement,
-      <HarmonyCacheProvider>
-        <SsrContextProvider
-          value={{
-            isServerSide: false,
-            isSsrEnabled: true,
-            pageProps,
-            isMobile,
-            history: null
-          }}
-        >
-          <Root />
-        </SsrContextProvider>
-      </HarmonyCacheProvider>
+
+      <ClientOnly load={() => import('./RootWithProviders')} fallback={null}>
```

Comment:
this is cool!

Reply:
Would this cause hydrate to fail, ie there isn't an html match from server -> client, since the initial render would just be "null"? Are we saying this is okay for the web scraper?

Reply:
For a crawler, this would never load so i dont think there would be a problem? The fallback to null might be a problem for an actual client though so i'll check that out

Reply:
so when a crawler hits the app it gets the html payload, but it's basically a browser, so doesn't it then run onRenderClient, meaning they will eventually import the full RootWithProviders?

Reply:
The idea here is to exclude the JS when `HYDRATE_CLIENT` is true, but I think this may not be the best way to achieve that.

`ClientOnly` doesn't really make sense in this context because `onRenderClient` already only renders on the client. So really it's just being use to do the lazy load which would be better done as:

```ts
if (HYDRATE_CLIENT) {
    const RootWithProviders = lazy(() => import('./RootWithProviders'))
    hydrateRoot(
      document.getElementById('root') as HTMLElement,
      <Suspense>
        <RootWithProviders>
      ...
```

But then getting to @dylanjeffers's point, in the case of `HYDRATE_CLIENT=true` this will (i think) fail hydration because the initial render won't match the html. If we are only using ssr for crawlers right now, maybe it's better to just delete `onRenderClient` (or rename it) so we don't do any client side JS

Reply:
ah sorry i should have mentioned. its not in this PR, the assumption is that we will turn off HYDRATE_CLIENT for the crawlers in the future so this all code in the `if (HYDRATE_CLIENT) {` will not get run

Reply:
Gotcha, so in this case could we just render null and call it a day? Or this is the way to toggle between the two?